tfhe/src/integer/server_key/radix_parallel/scalar_add.rs

```diff
@@ -1,8 +1,50 @@
-use crate::integer::block_decomposition::DecomposableInto;
+use crate::core_crypto::prelude::UnsignedNumeric;
+use crate::integer::block_decomposition::{BlockDecomposer, DecomposableInto};
 use crate::integer::ciphertext::IntegerRadixCiphertext;
-use crate::integer::ServerKey;
+use crate::integer::{BooleanBlock, RadixCiphertext, ServerKey};
 
 impl ServerKey {
+    pub fn unsigned_overflowing_scalar_add_assign_parallelized<Scalar>(
+        &self,
+        lhs: &mut RadixCiphertext,
+        scalar: Scalar,
+    ) -> BooleanBlock
+    where
+        Scalar: UnsignedNumeric + DecomposableInto<u8>,
+    {
+        if !lhs.block_carries_are_empty() {
+            self.full_propagate_parallelized(lhs);
+        }
+
+        self.unchecked_scalar_add_assign(lhs, scalar);
+        let overflowed = self.unsigned_overflowing_propagate_addition_carry(lhs);
+
+        let num_scalar_block =
+            BlockDecomposer::with_early_stop_at_zero(scalar, self.key.message_modulus.0.ilog2())
+                .count();
+
+        if num_scalar_block > lhs.blocks.len() {
+            // Scalar has more blocks so addition counts as overflowing
+            BooleanBlock::new_unchecked(self.key.create_trivial(1))
```

Comment:
care to elaborate why this is required, given we already have an overflow block, I'm guessing the block can be wrong ?

Reply:
Our scalar addition signature allows any clear integer type, so I can be adding a u16 to a radixciphertext with 8 bits
The way it works it that it truncates the u16 to 8bits, ignoring the rest, which is correct, but in the overflowing version
we have to make sure the ignored blocks are non zero, if any of the ignored block is non zero it means we tried to add a value that is bigger than the ciphertext max to the ciphertext, so we count that as overflow

Reply:
ah ok thanks and the LSB blocks may not be overflowing